### PR TITLE
Fix export/set/unset environment variables for fish shell when activating a new env

### PR DESCRIPTION
**Pull request by JafarAbdi**
I was running into an issue where activating an environment isn't working for the fish shell and it was messing up the `PATH` environment variable 

```
PATH=/home/jafar/micromamba/bin:/home/jafar/micromamba/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:set:-gx:CONDA_PREFIX:/home/jafar/micromamba:set:-gx:CONDA_SHLVL:1:set:-gx:CONDA_DEFAULT_ENV:base:set:-gx:CONDA_PROMPT_MODIFIER:(base)
```
as you can see from the variable above, it was being interpreted as appending to the list rather than a command, I fixed it by adding a semicolon after each command

The second commit renamed `MAMBA_PROMPT_MODIFIER` to `CONDA_PROMPT_MODIFIER`, grepping the repo I don't see other usages for `MAMBA_PROMPT_MODIFIER` and looking at the codebase `CONDA_PROMPT_MODIFIER` is the variable being set

Ubuntu version: `20.04`
Fish version `3.5.1`

### CODE REVIEW

**Comment by jonashaag** [comment]
Thanks for the patch! I don't understand how this fixes the problem. Shouldn't `\n` suffice to terminate a fish command?

**Comment by JafarAbdi** [comment]
> Shouldn't \n suffice to terminate a fish command?

That's what I thought, too, but it doesn't seem to be the case. I'll work on providing an easily reproducible example

**Comment by wolfv** [comment]
I guess it wouldn't hurt either, right @jonashaag ?

**Comment by jonashaag** [comment]
Yes but I wonder what's the underlying problem here. I'm worried that we're not actually fixing the problem.

**Comment by JafarAbdi** [comment]
Hmm, I can't reproduce it using a fish shell directly, maybe it's related to how CLI11 is executing those commands in https://github.com/mamba-org/mamba/blob/main/libmamba/src/api/shell.cpp#L150

File: libmamba/src/core/activation.cpp [hunk 1]
```diff
@@ -1204,17 +1204,17 @@ namespace mamba
 
         for (const std::string& uvar : env_transform.unset_vars)
         {
-            out << "set -e " << uvar << "\n";
+            out << "set -e " << uvar << ";\n";
```
Comment (JafarAbdi) [inline libmamba/src/core/activation.cpp:1207]: Should I remove the newline? It's redundant here!
```suggestion
            out << "set -e " << uvar << ";";
```

Comment (wolfv) [reply libmamba/src/core/activation.cpp:1207]: It probably looks cleaner with a newline, no?

**Comment by jonashaag** [comment]
Thanks for your contribution but I'm against merging this without understanding the problem, sorry!

**Comment by jonashaag** [comment]
Cherry-picked changes in https://github.com/mamba-org/mamba/pull/2101

**Pull request closed without merging**